File: typescript/packages/bench/src/rules/warrantyWindowRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class WarrantyWindowRule implements RuleContract {
    readonly key = "warranty-window";

    evaluate(data: Dataset): boolean {
        return data.devices.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/warranty_window_rule.rs
//! Rule 3/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct WarrantyWindowRule;

impl RuleContract for WarrantyWindowRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.devices.is_empty()
    }
}
